test/reg_sort_test.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: ASCII-8BIT
+require 'test_helper'
+require 'pwnlib/reg_sort'
+
+class RegSortTest < MiniTest::Test
+  include ::Pwnlib::RegSort::ClassMethod
+
+  def test_check_cycle
+    assert_equal([], check_cycle('a', a: 1))
+    assert_equal(['a'], check_cycle('a', a: 'a'))
+    assert_equal([], check_cycle('a', a: 'b', b: 'c', c: 'b', d: 'a'))
+    assert_equal(%w(a b c d), check_cycle('a', a: 'b', b: 'c', c: 'd', d: 'a'))
+  end
+
+  def test_regsort
+    regs = %w(a b c d x y z)
+    # normal
+    assert_equal([['mov', 'a', 1], ['mov', 'b', 2]], regsort({ a: 1, b: 2 }, regs))
+    # tmp
+    assert_equal([%w(mov X a), %w(mov a b), %w(mov b X)], regsort({ a: 'b', b: 'a' }, regs, tmp: 'X'))
+    # post move
+    assert_equal([['mov', 'a', 1], %w(mov b a)], regsort({ a: 1, b: 1 }, regs))
+    assert_equal([%w(mov c a), ['mov', 'a', 1], %w(mov b a)], regsort({ a: 1, b: 1, c: 'a' }, regs))
+    # resolve dependencies
+    assert_equal([%w(mov b a), ['mov', 'a', 1]], regsort({ a: 1, b: 'a' }, regs))
```

Comment:
Also add failing case for current version :P 
`regsort({a:'b', b:'a', c:'d',d:'c'}, %w(a b c d), tmp:'c')`